### PR TITLE
[windows|toolchain] Try using /Zi

### PR DESCRIPTION
**Pull request by Neumann-A**
Just testing.

Good example why `VCPKG_POST_PORTFILE_INCLUDES` would be required

Current status:
+ Make based ports are a bit screwed and have to restart without parallelism since there is no link job limit
+ maybe make based ports can be saved by /Fd and some automatic make variable.

### CODE REVIEW

File: scripts/toolchains/windows.cmake [hunk 1]
```diff
@@ -57,16 +57,38 @@ if(NOT _CMAKE_IN_TRY_COMPILE)
...
     unset(CHARSET_FLAG)
 
-    set(CMAKE_CXX_FLAGS_DEBUG "/D_DEBUG ${VCPKG_CRT_LINK_FLAG_PREFIX}d /Z7 /Ob0 /Od /RTC1 ${VCPKG_CXX_FLAGS_DEBUG}" CACHE STRING "")
-    set(CMAKE_C_FLAGS_DEBUG "/D_DEBUG ${VCPKG_CRT_LINK_FLAG_PREFIX}d /Z7 /Ob0 /Od /RTC1 ${VCPKG_C_FLAGS_DEBUG}" CACHE STRING "")
-    set(CMAKE_CXX_FLAGS_RELEASE "${VCPKG_CRT_LINK_FLAG_PREFIX} /O2 /Oi /Gy /DNDEBUG /Z7 ${VCPKG_CXX_FLAGS_RELEASE}" CACHE STRING "")
-    set(CMAKE_C_FLAGS_RELEASE "${VCPKG_CRT_LINK_FLAG_PREFIX} /O2 /Oi /Gy /DNDEBUG /Z7 ${VCPKG_C_FLAGS_RELEASE}" CACHE STRING "")
+    set(CMAKE_CXX_FLAGS_DEBUG "/D_DEBUG ${VCPKG_CRT_LINK_FLAG_PREFIX}d /Zi /FS /Ob0 /Od /RTC1 ${VCPKG_CXX_FLAGS_DEBUG}" CACHE STRING "")
+    set(CMAKE_C_FLAGS_DEBUG "/D_DEBUG ${VCPKG_CRT_LINK_FLAG_PREFIX}d /Zi /FS /Ob0 /Od /RTC1 ${VCPKG_C_FLAGS_DEBUG}" CACHE STRING "")
+    set(CMAKE_CXX_FLAGS_RELEASE "${VCPKG_CRT_LINK_FLAG_PREFIX} /O2 /Oi /Gy /DNDEBUG /Zi /FS ${VCPKG_CXX_FLAGS_RELEASE}" CACHE STRING "")
+    set(CMAKE_C_FLAGS_RELEASE "${VCPKG_CRT_LINK_FLAG_PREFIX} /O2 /Oi /Gy /DNDEBUG /Zi /FS ${VCPKG_C_FLAGS_RELEASE}" CACHE STRING "")
```
Comment (JackBoosY) [inline scripts/toolchains/windows.cmake:63]: I remember @ras0219-msft said the reason why we choose /Z7 here.

**Comment by JackBoosY** [comment]
https://dev.azure.com/vcpkg/public/_build/results?buildId=74925&view=results

**Comment by Neumann-A** [comment]
> Some form of design description of how the PDB name is set in such a way that ports won't stomp on each other; note that this is not the default.

solved by setting the name explicitly to the portname which is unique

> Some form of write-up on how downstream build systems (e.g. cmake using scripts/buildsystems/vcpkg.cmake or the msbuild integration or what have you) are supposed to get the right PDBs and make them available for their customers

solved. just have the pdbs where the dlls/libs are

> (probably) Some writeup / mechanism to avoid having ports affected by an already running mspdbsrv.

whatever that means?

**Comment by Neumann-A** [comment]
```
compile-c-c++ E:\vcpkg_folders\master2\buildtrees\boost-atomic\x64-windows-dbg\boost\build\17a8c9ca3b8927c9ea3f9b7cab707861\lock_pool.obj

    call "E:\vcpkg_folders\master2\buildtrees\boost-atomic\x64-windows-dbg\boost\standalone\msvc\943897acc6555446e7558f53bf1a3e85\msvc-setup.bat" amd64 >nul
 cl "..\src\lock_pool.cpp" -c -Fo"E:\vcpkg_folders\master2\buildtrees\boost-atomic\x64-windows-dbg\boost\build\17a8c9ca3b8927c9ea3f9b7cab707861\lock_pool.obj"     -TP /wd4675 /EHs /GR /Zc:throwingNew -nologo -DWIN32 -D_WINDOWS -W3 -utf-8 -GR -EHsc -MP -D_DEBUG -MDd -Zi -FS -Ob0 -Od -RTC1 /Z7 /Od /Ob0 /W3 /MDd /Zc:forScope /Zc:wchar_t /Zc:inline /favor:blend -DBOOST_ALL_NO_LIB=1 -DBOOST_ATOMIC_DYN_LINK=1 -DBOOST_ATOMIC_SOURCE -DBOOST_ATOMIC_USE_SSE2 -DBOOST_ATOMIC_USE_SSE41 -DBOOST_USE_WINDOWS_H "-I..\include" "-IE:\vcpkg_folders\master2\installed\x64-windows\include" "-I..\src" 

cl : Command line warning D9025 : overriding '/Zi' with '/Z7'
```
also boost-build seems to add its own flags (all prefixed with `/`): @ras0219-msft

**Comment by Neumann-A** [comment]
current status:
```
     Failed: 31 (1.71%)
    Passed: 1,740 (95.81%)
    Other: 45 (2.48%)
    Total: 1,816 
```

**Comment by Neumann-A** [comment]
Hmm:
```
SUMMARY FOR x64-windows
    SUCCEEDED: 1792
    BUILD_FAILED: 15
    FILE_CONFLICTS: 1
    CASCADED_DUE_TO_MISSING_DEPENDENCIES: 11
```

To fix (buildtree in pdb path)
 - [ ] yasm
 - [ ] python2
 - [ ] botan
 - [ ] chakracore
 - [ ] crashpad
 - [ ] halide
 - [ ] qwt
 - [ ] libqpg
 - [ ] kf5auth
 - [ ] kf5i18n
 - [ ] kf5iconthemes
 - [ ] kf5windowsystem
 - [ ] kf5sonnet
 - [ ] kf5textwidgets
 - [ ] kf5kio
 - [ ] kf5plotting
 - [ ] libgit2
 - [ ] libproxy
 - [ ] libqglviewer
 - [ ] luajit
 - [ ] mdnsresponder
 - [ ] openni2
 - [ ] pmdk
 - [ ] qca
 - [ ] qcustomplot
 - [ ] qscintilla
 - [ ] unicorn
 - [ ] unrar

(most are probably qmake ports. )

93 matches for pdb files not found 
 - [ ] boost-atomic
 - [ ] icu
 - [ ] boost-regex
 - [ ] ace
 - [ ] boost-system
 - [ ] boost-chrono
 - [ ] boost-container
 - [ ] boost-date-time
 - [ ] boost-thread
 - [ ] boost-serialization
 - [ ] pthreads
 - [ ] boost-context
 - [ ] openblas
 - [ ] boost-filesystem
 - [ ] boost-random
 - [ ] boost-stacktrace
 - [ ] boost-math
 - [ ] boost-locale
 - [ ] gettext
 - [ ] mp3lame
 - [ ] boost-log
 - [ ] boost-program-options
 - [ ] boost-timer
 - [ ] boost-iostreams
 - [ ] boost-test
 - [ ] libusb
 - [ ] azure-kinect-sensor-sdk
 - [ ] vcpkg-tool-lessmsi/python2
 - [ ] boost-contract
 - [ ] boost-fiber
 - [ ] boost-graph
 - [ ] python2
 - [ ] python3
 - [ ] boost-python
 - [ ] boost-mpi
 - [ ] boost-graph-parallel
 - [ ] boost-json
 - [ ] boost-nowide
 - [ ] boost-type-erasure
 - [ ] boost-wave
 - [ ] tbb
 - [ ] clockutils
 - [ ] cocoyaxi
 - [ ] octomap
 - [ ] directx-dxc
 - [ ] directxtk12
 - [ ] discord-game-sdk
 - [ ] fftw3
 - [ ] dstorage
 - [ ] dxsdk-d3dx
 - [ ] easyhook
 - [ ] ecsutil
 - [ ] forge
 - [ ] g2o
 - [ ] geogram
 - [ ] gpgmm
 - [ ] hunspell
 - [ ] libirecovery
 - [ ] libosmscout
 - [ ] libspatialindex
 - [ ] mimalloc
 - [ ] miniz
 - [ ] mokeys-audio
 - [ ] ngspice
 - [ ] vcpkg-tool-mozbuild
 - [ ] ocilib
 - [ ] onnxruntime
 - [ ] opencc
 - [ ] zydis
 - [ ] rendergraph
 - [ ] rtabmap
 - [ ] webview2
 - [ ] scintilla
 - [ ] sciter-js
 - [ ] sdl1
 - [ ] xaudio2redist
 - [ ] zero-ice

**Pull request closed without merging**